### PR TITLE
Feature brenach

### PR DESCRIPTION
**Pull request by SunnySparks**
This is a reference from https://github.com/SunnySparks/re-former/pull/4

In this pull request we:

Created the HTML form and database models
Created the form_tag and form_for using rails
Enabled edition and created the proper views for said forms

### CODE REVIEW

**Comment by SunnySparks** [comment]
Dear TSE: 
We have been working on trying to solve the Tests/RSpec issue that is displayed since Monday, we are getting lost on this one because we have tried different solutions (Like reinstalling Ruby, erasing the cache, updating the gemfile, etc) with no results. We would appreciate it if you have any hint or suggestion to find a solution to such issue.

**Review by kevcap — CHANGES_REQUESTED** [review]
# [STATUS] CHANGES REQUESTED ♻

### Hey, nice project so far. I noticed that you guys are missing some key features of this project, so we have to improve a few things before moving on:

- [ ] You have linter errors;
- [x] You have commented code on your users_controller.rb, line 8;
On this project your application should do the following:
- [ ] User can sign in ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- [ ] As a signed-in user you can create new post ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_post_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- [ ] As a signed-in user you can see post's author ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_see_author_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- [ ] User can sign_out ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_out_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- [ ] As a guest you can see only post's text ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

#### Authentication and User setup

- [ ] The application should use the `devise` gem for authentication ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/bcrypt_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

- [ ] The application should has Sign in/Sign out links present in any view of the app ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/))

- [ ] The User should has*many posts *(not explicity said in specification - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_associations_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication) but with extra hint - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/associations_hint_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/) )\_


#### Posts

##### routes

- [ ] The application routes should be created using `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

##### Post model

- [ ] It should have `belongs_to` a `user` _(not explicity said in specification - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_associations_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication) but with extra hint - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/associations_hint_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/) )_

##### PostsController

- [ ] It should use `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

About your question, I wasn't able to find the reason that is causing that error on the rspec.

## As always, I wish you all the very best and see you at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Comment by SunnySparks** [comment]
> # [STATUS] CHANGES REQUESTED ♻
> ### Hey, nice project so far. I noticed that you guys are missing some key features of this project, so we have to improve a few things before moving on:
> * [ ]  You have linter errors;
> * [x]  You have commented code on your users_controller.rb, line 8;
>   On this project your application should do the following:
> * [ ]  User can sign in ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
> * [ ]  As a signed-in user you can create new post ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_post_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
> * [ ]  As a signed-in user you can see post's author ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_see_author_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
> * [ ]  User can sign_out ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_out_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
> * [ ]  As a guest you can see only post's text ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
> 
> #### Authentication and User setup
> * [ ]  The application should use the `devise` gem for authentication ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/bcrypt_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
> * [ ]  The application should has Sign in/Sign out links present in any view of the app ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/))
> * [ ]  The User should has*many posts *(not explicity said in specification - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_associations_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication) but with extra hint - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/associations_hint_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/) )_
> 
> #### Posts
> ##### routes
> * [ ]  The application routes should be created using `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
> 
> ##### Post model
> * [ ]  It should have `belongs_to` a `user` _(not explicity said in specification - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_associations_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication) but with extra hint - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/associations_hint_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/) )_
> 
> ##### PostsController
> * [ ]  It should use `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
> 
> About your question, I wasn't able to find the reason that is causing that error on the rspec.
> 
> ## As always, I wish you all the very best and see you at the top🚀
> 👨🏻‍💻 **TSE: Kevin Alves**
> 
> 📫 Have any question? Better call Kevin:
> 
> [ gmail](mailto:kevinalves.js@gmail.com) [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

Dear TSE. I'm sorry about the confusion that was generated. As per the changes that are requested, we noticed that such changes belong to a different project, the confusion was generated on **our** end at the time of requesting the code review to a different project that we haven't arrived at yet.
we are kindly requesting a re-revision of our project, and if possible you could provide us some guidance about the Test / RSpec issue our project is facing (As mentioned before, we've been following instructions on the internet such as reinstalling Ruby, erasing the cache, updating the gemfile, etc with no solutions so far. We greatly appreciate your time and feedback.

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <h1> This is the Registration page </h1>
 
+
+
 <%= form_for(@user, url: "/users") do |f| %>
```
Comment (rindrajosia) [inline app/views/users/new.html.erb]: `<%= form_for @user do |f|%>` is enough here.

**Review by rindrajosia — APPROVED** [review]
**status: approved**
Great job @team,  I approve this project but before that, you need to make some changes before merging.

Please  read the comments below.

Happy coding
Rindra Josia.

**Comment by rindrajosia** [comment]
- [ ] Optional In your form_for, change `<%= f.submit %>` to `<%= f.submit :Submit%>`
- [ ] In your form_tag you don't need this :`<input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>">`
- [ ] In your form It's preferable to use `users_path` than using `/users`

If you want to run Rspec in your project:

1. Change ruby version in your `Gemfile` and `.ruby-version`
2. add this in you Gemfile:
```
group :development, :test do
  gem 'database_cleaner'
  gem 'capybara'
  gem 'factory_bot_rails'
  gem 'faker'
  gem 'launchy'
  gem 'rspec-rails'
end
```
```
group :test do
  gem 'database_cleaner-active_record'
  gem 'selenium-webdriver'
  # Easy installation and use of chromedriver to run system tests with Chrome
  gem 'chromedriver-helper'
end
```

3. Use this command to run test: `bundle exec rspec `

**Comment by SunnySparks** [comment]
> group :test do
>   gem 'database_cleaner-active_record'
>   gem 'selenium-webdriver'
>   # Easy installation and use of chromedriver to run system tests with Chrome
>   gem 'chromedriver-helper'
> end

Thank you for assisting us on the RSpec matter, I really appreciate it. I already made the requested changes as well

**Pull request merged**